### PR TITLE
 Allow signing on VS 2019

### PR DESCRIPTION
**Pull request by davkean**
This allows signing on VS 2019, instead of limiting to VS 2017.

Originally we were going to consume this by upgrading to the latest version of RoslynTools.RepoToolset. I spent 2 days on that port and gave up; it's a significant work item due to a large number of breaking changes. I timeboxed and ran out of time.

Instead, I've gone down the approach of copying the Sign.proj from the toolset, and slightly changing it. Both are in two separate commits.

### CODE REVIEW

File: build/proj/Sign.proj [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  <Import Project="..\import\Versions.props" />
+  <Import Project="..\import\NuGet.props" />
+  <Import Project="..\import\RepoLayout.props" />
+
+  <Target Name="Sign">
+
+    <PropertyGroup>
+      <!-- VS2019 and later place msbuild.exe under "MSBuild\Current", while VS2017 places it under "MSBuild\15.0". -->
+      <_MSBuildExePath Condition="Exists('$(VSINSTALLDIR)\MSBuild\Current\Bin\msbuild.exe')">$(VSINSTALLDIR)\MSBuild\Current\Bin\msbuild.exe</_MSBuildExePath>
+      <_MSBuildExePath Condition="'$(_MSBuildExePath)' == ''">$(VSINSTALLDIR)\MSBuild\15.0\Bin\msbuild.exe</_MSBuildExePath>
```
Comment (davidwengier) [inline build/proj/Sign.proj]: I think you can remove the line above, and change this to use `$(MSBuildToolsVersion)` to be simpler and work for both versions...
```suggestion
      <_MSBuildExePath Condition="'$(_MSBuildExePath)' == ''">$(VSINSTALLDIR)\MSBuild\$(MSBuildToolsVersion)\Bin\msbuild.exe</_MSBuildExePath>
```

Comment (davkean) [reply build/proj/Sign.proj]: Much cleaner.

File: build/proj/Sign.proj [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Project DefaultTargets="Sign">
+  <!--
+    Required parameters:
+      ConfigFile                    SigntToolData.json path.
```
Comment (drewnoakes) [inline build/proj/Sign.proj]:
```suggestion
      ConfigFile                    SignToolData.json path.
```

**Comment by drewnoakes** [comment]
Does this work on VS2017 too?

**Comment by davkean** [comment]
MSBuildToolsVersion should do so.

**Comment by davkean** [comment]
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=2945280 is the sign build that is verifying this change.

**Pull request merged**